spec/ics-027-direct-interchain-exchange/ReadMe.md

```diff
@@ -0,0 +1,74 @@
+
+This order persists on FooChain (in the form of locked tokens) and BarChain (in the form of an open limit order) until one of two things happen.
+
+Case A: Alice cancels the order by submitting a cancellation order on BarChain: {type:"limit-cancel", pair:"bar/foo", sell:"20foo", account:"Alice"}.
+Case B: Bob (say) partially or fully matches the order by submitting a limit order on BarChain: {type:"limit", pair:"bar/foo", sell:"100bar", price:"0.8"}.
+
+There are ways to design the protocol such that Alice can cancel the order without having to submit an order to BarChain, but due to the asynchrony of IBC it cannot be immediate for both Alice and Bob simultaneously.  Such extensions to the interchain exchange protocol are not described here.
+
+For both case A and case B, a corresponding opposite IBC message is created from BarChain to FooChain.  In case A, the 20 Foo tokens are released back to Alice.  In case B, the 20 Foo tokens are released back to Bob, and in this example (since Bob's order volume was greater than Alice's), a new partial limit order would be created on FooChain on behalf of Bob.
+
+#### Delivery
+
+While the delivery of matched orders may be routed anywhere, by default we propose that the delivery happens on the Hub as IBC pegged tokens.
+It simplifies custody for Alice and Bob because of common custodial protocols on the Hub.
+It creates a substantial social pressure toward the economic finality of the exchange of tokens by having them delivered to Hub accounts.
+Only delivered tokens need to be on the Hub, not the tokens behind pending limit orders, so as long as users move their funds out of the Hub soon after execution, the security requirement on the hub is minimal.
+
+#### Hub Collateralization
+
+With N^2 IBC connections between N zones for this Direct Interchain Exchange Protocol (DIEP), it becomes difficult for traders and market makers to make sense of the complexity arising from chain security.
+Additionally, by forcing a chain to fail with a double spend attack (a consensus fork), it may become profitable to induce failure, especially with more complex scenarios that require consistency across the interchain.  For example, interchain pair trading may become profitable to attack.
+
+To mitigate such attacks and guarantee more quantifiable skin in the game for the validators, the Hub should play the role of interchain collateralization.  All zones in the DIEP network should connect via IBC to the Hub, and the Hub should be modified to allow staking on behalf of these zones.  ATOM tokens staked on behalf of other zones should not be subject to the full inflationary ATOM tax (e.g. these staked zones may earn a proportionate amount of inflationary ATOMs, less any fees).  And the Hub becomes the arbiter for deciding whether a zone is consistent, or whether it had been double-spend attacked.  Any slashed ATOMs may be used to compensate the victims of an attack, although some ATOMs should probably always be burned.
```

Comment:
I'm a bit confused, so in this model an _entire zone_ can actually be staked on the Hub? Or simply a validator _on_ another zone can also act individually as a sort of "intermediary" validator? (this should be more explicitly worded within this paragraph). 

If you were referring to an entire zone being able to stake with the hub it introduces some potentially unnecessary complexity because now those inflationary rewards provided to the "zone-validator" must be redistributed somehow within the zone (base on whoever fronted the original atoms there).

Reply:
entire zone can actually be staked on the Hub? Or simply a validator on another zone can also act individually as a sort of "intermediary" validator?

too much risk for 1 validator imo. probably some percentage of validators (say 2/3 of all stake) is needed.

Reply:
> If you were referring to an entire zone being able to stake with the hub it introduces some potentially unnecessary complexity because now those inflationary rewards provided to the "zone-validator" must be redistributed somehow within the zone (base on whoever fronted the original atoms there).

yes. I call it "IBC cross-chain distribution protocol". The interchain collaterization will definitely need this additional protocol to transfer rewards from zone to the hub. Definitely "a validator" case is not intended in this context I guess. As melekes said, it is not reasonable to do so from security-wise perspective.